solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,492 @@
+    error_message = "'existing_cos_instance_crn' is required if 'enable_vpc_flow_logs' is set to true."
+  }
+}
+
+variable "cos_bucket_name" {
+  description = "Name of the Cloud Object Storage bucket to be created collect VPC flow logs."
+  type        = string
+  default     = "cos-bucket"
+}
+
+variable "kms_encryption_enabled_bucket" {
+  description = "Set to true if Cloud Object Storage bucket needs to be KMS encryption enabled."
+  type        = bool
+  default     = false
+
+  validation {
+    condition     = !var.enable_vpc_flow_logs ? (var.kms_encryption_enabled_bucket ? false : true) : true
+    error_message = "'kms_encryption_enabled_bucket' should be false if 'enable_vpc_flow_logs' is set to false."
+  }
+
+  validation {
+    condition     = var.kms_encryption_enabled_bucket ? ((var.existing_kms_key_crn != null || var.existing_kms_instance_crn != null) ? true : false) : true
+    error_message = "Either 'existing_kms_key_crn' or 'existing_kms_instance_crn' is required if 'kms_encryption_enabled_bucket' is set to true."
+  }
+}
+
+variable "skip_cos_kms_auth_policy" {
+  type        = bool
+  description = "To skip creating an IAM authorization policy that allows Cloud Object Storage(COS) to access KMS key."
+  default     = false
+}
+
+variable "management_endpoint_type_for_bucket" {
+  description = "The type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets (`public`, `private`, or `direct`). If you are using a private endpoint, make sure that you enable virtual routing and forwarding (VRF) in your account, and that the Terraform runtime can access the IBM Cloud Private network."
+  type        = string
+  default     = "public"
+  validation {
+    condition     = contains(["public", "private", "direct"], var.management_endpoint_type_for_bucket)
+    error_message = "The specified `management_endpoint_type_for_bucket` is not valid. Specify a valid type of endpoint for the IBM Terraform provider to use to manage Cloud Object Storage buckets."
+  }
+}
+
+variable "cos_bucket_class" {
+  type        = string
+  default     = "standard"
+  description = "The storage class of the newly provisioned Cloud Object Storage bucket. Specify one of the following values for the storage class: `standard`, `vault`, `cold`, `smart` (default), or `onerate_active`."
+  validation {
+    condition     = contains(["standard", "vault", "cold", "smart", "onerate_active"], var.cos_bucket_class)
+    error_message = "Specify one of the following values for the `cos_bucket_class`: `standard`, `vault`, `cold`, `smart`, or `onerate_active`."
+  }
+}
+
+###############################################################################################################
+# KMS
+###############################################################################################################
+
+variable "existing_kms_key_crn" {
```

Comment:
I actually suggest you name this `existing_flow_logs_bucket_kms_key_crn` so it is clear what the key is used for. Maybe some day there be something else in this DA that will need its own KMS key, so better to future proof it

Reply:
updated